### ISSUE TITLE
Step is not stepping consistently

### ISSUE THREAD

**Issue by shaleh** [derekparker/delve#966]
Delve Debugger
Version: 1.0.0-rc.1
Build: $Id: b6077a6a35c48a31716aad2ad3fdf77bbdf70abd $

go version go1.8.3 linux/amd64

Ubuntu Xenial, AMD64

As a user, I am seeing times where `step` does not step.

I have a breakpoint set. The function in question is a recursive function. I have set a condition to catch just before the end of the recursion to diagnose an edge case. The condition is triggered and the break occurs. After inspecting the state I attempt to use `step` to continue the recursion. But instead it acts like a `next`. I have been able to call `step` a couple of times and then it misbehaves.

The best work around I have so far is adjusting the condition.

**Comment by aarzilli** [comment]
Duplicate of #844?

**Comment by shaleh** [comment]
It is related. In my case I get a step that the conditional would not trigger. But it fails on the next step or the one after.

The piece I was debugging had a memory overlap bug. The recursion passed an updated slice in like this:

    result, err = compute(args...., append(current, newValue))

**Comment by aarzilli** [comment]
You'd have to be more specific. What are you stepping on when it's failing? Where's the conditional breakpoint set?

**Comment by shaleh** [comment]
I have a breakpoint on definition line of  `compute`. I have a conditional on that break point set to check one of the parameters. When execution hits the conditional and stops I am able to use `next` and `print` to look around. I can use `step` to go another level or two deeper into the recursion. The parameter being checked by the conditional is changing. But I only get a step or maybe two before `step` acts like `next` and the execution of that recursive branch completes.

**Comment by aarzilli** [comment]
This sounds like an exact duplicate of #844. The times when step manages to step into compute are just because the condition happens to be true, it will never work otherwise.

**Comment by shaleh** [comment]
I believe it was failing while the condition should not have passed but only once. The bug I was hunting was resolved after I submitted the ticket. Let me see if I can roll back the check in and verify this.

**Comment by aarzilli** [comment]
> I believe it was failing while the condition should not have passed but only once

If the setup is as you have described it will fail every single time if the condition isn't met. Note that since you have a breakpoint on the definition you could also end up deeper into the recursion than you expected.

**Comment by shaleh** [comment]
Deeper into the recursion than expected is the right answer. It is failing as #844 says. I will close this.

Thank you for the time. Sorry if I wasted it.

**Comment by aarzilli** [comment]
No problem. I'll see if I can prioritize #844 since there's somebody other than me using conditional breakpoints.
